### ISSUE TITLE
Intermittent flashes during playback

### ISSUE THREAD

**Issue by ZereoX** [tvdzwan/hyperion#284]
Issue:
During video playback, a section or all LEDs will intermittently flash white. Configuration file and  video linked below.

Note: This issue occurs only when using the Frame Grabber (i.e Hyperion-remote/Android app mood lighting works without any issues).

Information:
- Raspbmc version: (raspbmc-rls-1.0-hardfp-b20130208-u20150125)
- Kodi Version: Kodi (14.1 Git:2015-02-02-81c4046-dirty)
- Kernel Version: Linux raspbmc 3.12.31 #2 PREEMPT Wed Oct 29 09:24:56 UTC 2014 armv6l GNU/Linux
- Configuration File: [hyperion.config.json](http://pastebin.com/Rzxtyivv)
- Video: www.youtube.com/watch?v=L6NM-Kr8Vzs

Edit:

Oopss. @timk95, the video should be public now.

**Comment by timk95** [comment]
I'm facing the exact same issue with the Frame Grabber. My configuration is roughly the same as yours. I've tried to change the baud rate and the frequency of the frame grabber but it doesn't change anything.

When I play the swirl effect in the menu it works just fine and smooth but when I activate it during playback it starts to flash white (and sometime other colors) aswel.

@ZereoX you video is set to private, we can't see it.

**Comment by tvdzwan** [comment]
Did you try other baudrates for the SPI? People have reported interference at certain baudrates during playback. This could cause this behaviour.

**Comment by timk95** [comment]
@tvdzwan I have the exact same issue. Changing the baud rate didn't work for me. Tried it from 10 up to 1.000.000. Didn't make it worse or better.

I'm using Rasplex 0.5.1 which is based on OpenElec. Forgot to mention that in my previous post.

**Comment by ZereoX** [comment]
@tvdzwan I can confirm @timk95's findings. I've tested a wide range of Baud Rates (115200, 25000, 500000...) and they all seem to incur this issue. As for the Playback Quality, I still see flashes even when the screen is still on the XBMC Menu.

**Comment by ntim** [comment]
Very rarely I also have these flashes with my Philips HUE system (only that they are blue due to color space transformations). 

Using OSMC RC (Kernel 3.18.9-5-osmc, Kodi 14.1) on RPi2

**Comment by timk95** [comment]
I've made a video of the flashes. https://www.dropbox.com/s/nv9krqm9r2dt4wy/Film%20op%2022-03-15%20om%2014.02.mov?dl=0

**Comment by vena** [comment]
i had issues with random flashes, as well.  turned out that the ground on the pi's GPIO pin running the LED strip's data line wasn't at the same potential as the LED strip power lines's even though it was all powered through the same PSU.  the pi was being powered by a USB hub powered by the PSU, and i was using that USB hub to also power an HDMI splitter and CVBS converter.  i suspect the hub was taking the ground off the HDMI connection and creating a ground loop.

explicitly grounding the pi from one of its ground pins to the PSU fixed the flicker and flashes.

**Comment by ZereoX** [comment]
Hey @vena, my Pi is powered by a Smart Phone charger while my LEDS are powered by a 10A PSU. If I understood correctly, connecting one of my Pi's ground GPIO to my PSU negative would do the trick?

**Comment by vena** [comment]
caveat - i am not an electrical engineer :)

i would probably want the pi and leds on the same power supply just to eliminate any possibility for funky ripple interaction, but essentially yes.  you want the leds' power and data at the same ground potential.

also, if your leds are WS2811/WS2812, you may need to run the data from the pi through a logic level shifter like a 74AHCT125 to get the voltage up to 5v as described here: https://learn.adafruit.com/neopixels-on-raspberry-pi

**Comment by ZereoX** [comment]
I will try to power my Pi through the 5v GPIO and have both my Pi and LEDS on the same PSU. If that works then I'll wire up a USB hub to properly power my Pi. 

Thanks @vena.

**Comment by vena** [comment]
i wouldn't bypass the micro usb port's protection circuitry if i could avoid it.  afaik, the gpio pins are unprotected.

it's easy enough to cut the end off a micro-usb cable and wire its power lines (they're usually red and black), and it could save you $25 replacing the pi.

**Comment by ZereoX** [comment]
After doing what @vena suggested, I split a USB cable and spliced my PSU to power both my LEDS and Pi. It seems to have alleviated the issue slightly. Flashes seem to be mush more infrequent.

**Comment by ZereoX** [comment]
I feel so stupid. I have fixed my issue and will be closing this ticket.

My mistake came in the words of "Led in top corner", "Led in bottom corner" and not being able to count. I was under the impression from reading the Wiki that it simply meant that a LED was positioned at each corner of your Monitor/Screen. Meaning each pair of LED strip met perpendicularly at the very corner of the screen. I was therefore under the assumption that setting those settings on/off would compensate for a potential gap at the corners between your strips.

My setup is a 21x34 (HxW) arrangement of WS2801, creating a perfect rectangle around my entire television about half an inch from the border. I therefore select true for both Top/Bottom LED corners. Hyperion therefore showed me that I ended up with 114 LEDS in my setup. Seems good.

It took until now for me to realized that I didn't have 114 LEDS. If I had bothered to count (21+34)x2 ≠ 114. I simply adjusted my configuration file to go from 0 to 109(110 LEDS) instead of 113 and voilà, no more flashes.

tl;dr: Toggling either of the corner led options will actually add 2 LEDS each to your total.

**Comment by timk95** [comment]
Would like to add that I as well managed to identify my problem with the flickering. It was the LAN cable. On WiFi it didn't happen anymore.
